imagebuildah/stage_executor.go

```diff
@@ -1903,6 +1903,10 @@ func (s *StageExecutor) getCreatedBy(node *parser.Node, addedContentSummary stri
 					continue
 				}
 				mountOptionSource = mountInfo.Source
+				mountOptionSource, err = imagebuilder.ProcessWord(mountOptionSource, s.stage.Builder.Arguments())
+				if err != nil {
+					return "", fmt.Errorf("getCreatedBy: while replacing arg variables with values for format %q: %w", mountOptionSource, err)
+				}
```

Comment:
Is there anything preventing other arguments to this flag, or other flags to this instruction, from also referencing build args?  If not, does the right thing happen when they do?

Reply:
Variables already expand correctly for other flags, issue happend to `source=` because I decided to use `source` directly from node without expanding it in this PR https://github.com/containers/buildah/pull/5691


I can open a PR which can add a test to verify various flags of `--mount` but wondering if that should a different PR or this one ? Since in this PR I only tested the argument which was passed on to `generatePathChecksum`

Reply:
If that's already covered by other test cases, there's no need for that.